resources/Microsoft.Windows.Assertion/Microsoft.Windows.Assertion.psm1

```diff
@@ -0,0 +1,461 @@
+    }
+
+    [void] Set()
+    {
+        # This resource is only for asserting the Processor Architecture requirement.
+    }
+}
+
+[DSCResource()]
+class HyperVisorPresent
+{
+    # Key required. Do not set.
+    [DscProperty(Key)]
+    [string]$SID
+
+    [DscProperty(Mandatory)]
+    [bool] $Required
+
+    [DscProperty(NotConfigurable)]
+    [bool] $HyperVisorPresent
+
+    [HyperVisorPresent] Get()
+    {
+        $this.HyperVisorPresent = Get-ComputerInfo | Select-Object -ExpandProperty HyperVisorPresent
+
+        return @{
+            Required          = $this.Required
+            HyperVisorPresent = $this.HyperVisorPresent
+        }
+    }
+
+    [bool] Test()
+    {
+        $currentState = $this.Get()
+        return $currentState.Required -eq $currentState.HyperVisorPresent
+    }
+
+    [void] Set()
+    {
+        # This resource is only for asserting the presence of a HyperVisor.
+    }
+}
+
+[DSCResource()]
+class OsInstallDate
+{
+    # Key required. Do not set.
+    [DscProperty(Key)]
+    [string]$SID
+
+    [DscProperty()]
+    [string] $Before = [System.DateTime]::Now
+
+    [DscProperty()]
+    [string] $After
+
+    [DscProperty(NotConfigurable)]
+    [string] $InstallDate
+
+    [OsInstallDate] Get()
+    {
+        # Try-Catch isn't a good way to do this, but `[System.DateTimeOffset]::TryParse($this.Before, [ref]$parsedBefore)` is erroring
+        try
+        {
+            $this.Before = $this.Before ? [System.DateTimeOffset]::Parse($this.Before) : $null
+        }
+        catch
+        {
+            throw "'$($this.Before)' is not a valid Date string."
+        }
+
+        # Try-Catch isn't a good way to do this, but `[System.DateTimeOffset]::TryParse($this.After, [ref]$parsedAfter)` is erroring
+        try
+        {
+            $this.After = $this.After ? [System.DateTimeOffset]::Parse($this.After) : $null
+        }
+        catch
+        {
+            throw "'$($this.After)' is not a valid Date string."
+        }
+
+        $this.InstallDate = [System.DateTimeOffset]::Parse($(Get-ComputerInfo | Select-Object -ExpandProperty OsInstallDate))
+
+        return @{
+            Before      = $this.Before
+            After       = $this.After
+            InstallDate = $this.InstallDate
+        }
+    }
+
+    [bool] Test()
+    {
+        $currentState = $this.Get()
+        # this.Get() should always return [System.DateTimeOffset] or $null which can be compared directly
+        # $null should always be treated as less than a [System.DateTimeOffset]
+        return ($currentState.InstallDate -gt $currentState.After) -and ($currentState.InstallDate -lt $currentState.Before)
+    }
+
+    [void] Set()
+    {
+        # This resource is only for asserting the OS Install Date.
+    }
+}
+
+# This is the same function from Microsoft.Windows.Developer, just included here as it seemed to make sense
+[DSCResource()]
+class OsVersion
```

Comment:
Should we actually include it? DSC v3 already has osinfo, see: https://github.com/PowerShell/DSC/tree/main/osinfo

Reply:
Not everyone is going to move to DSCv3 right away, and currently WinGet only really supports DSCv2. I know that this resource is / will be duplicated, but I figure that it's easy enough to add a deprecation warning, or perhaps even call the DSCv3 method internally, once DSCv3 is officially released and is more widely supported